cmd/kubeadm/app/util/config/joinconfiguration.go

```diff
@@ -60,54 +60,76 @@ func SetJoinControlPlaneDefaults(cfg *kubeadmapi.JoinControlPlane) error {
 // Right thereafter, the configuration is defaulted again with dynamic values (like IP addresses of a machine, etc)
 // Lastly, the internal config is validated and returned.
 func JoinConfigFileAndDefaultsToInternalConfig(cfgPath string, defaultversionedcfg *kubeadmapiv1beta1.JoinConfiguration) (*kubeadmapi.JoinConfiguration, error) {
```

Comment:
I know this is not strictly in scope of this PR, but what about changing `JoinConfigFileAndDefaultsToInternalConfig` into `LoadOrDefaultJoinConfig` or simply `GetJoinConfig`?

Reply:
`LoadOrDefaultJoinConfig` SGTM